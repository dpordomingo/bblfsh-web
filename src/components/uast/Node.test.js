import React from 'react';
import { shallow, mount } from 'enzyme';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore();

import {
  Node,
  CollapsibleItem,
  StyledCollapsibleContent,
  StyledCollapsibleTitle,
  StyledItem,
  Roles,
  Children,
  Property,
  Properties,
  Value,
  PropertyName,
  Position,
} from './Node';

const shouldMatchSnapshot = comp => {
  const wrapper = renderer.create(comp);
  expect(wrapper.toJSON()).toMatchStyledComponentsSnapshot();
};

describe('CollapsibleItem', () => {
  it('collapsed item should be hidden', () => {
    const wrapper = mount(<CollapsibleItem collapsed={true} />);
    const content = wrapper.find(StyledCollapsibleContent);
    expect(content).toHaveStyleRule('display', 'none');
  });

  it('expanded item should not be hidden', () => {
    const wrapper = mount(<CollapsibleItem collapsed={false} />);
    const content = wrapper.find(StyledCollapsibleContent);
    expect(content).toHaveStyleRule('display', 'block');
  });

  it('is collapsed when the title is clicked', () => {
    const wrapper = mount(<CollapsibleItem collapsed={false} />);
    const title = wrapper.find(StyledCollapsibleTitle);
    title.simulate('click');
    const content = wrapper.find(StyledCollapsibleContent);
    expect(content).toHaveStyleRule('display', 'none');
  });

  it('is expanded when the title is clicked and it was collpased', () => {
    const wrapper = mount(<CollapsibleItem collapsed={true} />);
    const title = wrapper.find(StyledCollapsibleTitle);
    title.simulate('click');
    const content = wrapper.find(StyledCollapsibleContent);
    expect(content).toHaveStyleRule('display', 'block');
  });

  it('calls onNodeSelected when the mouse is over the node', () => {
    const spy = jest.fn();
    const wrapper = mount(<CollapsibleItem onNodeSelected={spy} />);
    wrapper.find(StyledItem).simulate('mousemove');

    expect(spy.mock.calls.length).toBe(1);
  });

  it('renders correctly', () => {
    shouldMatchSnapshot(
      <CollapsibleItem name="foo">yada yada</CollapsibleItem>
    );
  });
});

test('Roles renders correctly', () => {
  shouldMatchSnapshot(<Roles roles={['a', 'b', 'c']} />);
});

test('Position renders correctly', () => {
  shouldMatchSnapshot(
    <Position name="foo" position={{ Offset: 1, Line: 1, Col: 1 }} />
  );
});

describe('Children', () => {
  const initialState = {
    code: {
      ast: {
        2: {
          id: 2,
          StartPosition: { Col: 1, Line: 1, Offset: 1 },
          EndPosition: { Col: 1, Line: 1, Offset: 1 },
          Roles: ['a', 'b'],
          InternalType: 'foo',
        },
        3: {
          id: 3,
          StartPosition: { Col: 2, Line: 1, Offset: 1 },
          EndPosition: { Col: 2, Line: 1, Offset: 1 },
          Roles: ['a', 'c'],
          InternalType: 'bar',
        },
      },
    },
    options: { showLocations: false },
  };
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('renders correctly', () => {
    const children = [2, 3];
    shouldMatchSnapshot(
      <Provider store={store}>
        <Children ids={children} />
      </Provider>
    );
  });
});

test('Property renders correctly', () => {
  shouldMatchSnapshot(<Property name="foo" value="bar" />);
});

test('Properties renders correctly', () => {
  shouldMatchSnapshot(
    <Properties
      properties={{
        foo: 'bar',
        baz: 'qux',
      }}
    />
  );
});

describe('Value', () => {
  it('should have text "null" when the value is null', () => {
    const wrapper = mount(<Value value={null} />);
    expect(wrapper.text()).toBe('null');
  });

  it('should render correctly with value', () => {
    shouldMatchSnapshot(<Value value={1} />);
  });
});

test('PropertyName renders correctly', () => {
  shouldMatchSnapshot(<PropertyName name="foo" />);
});

describe('Node', () => {
  const initialState = {
    code: {
      ast: {
        2: {
          id: 2,
          StartPosition: { Col: 1, Line: 1, Offset: 1 },
          EndPosition: { Col: 1, Line: 1, Offset: 1 },
          Roles: ['a', 'b'],
          InternalType: 'foo',
        },
        3: {
          id: 3,
          StartPosition: { Col: 2, Line: 1, Offset: 1 },
          EndPosition: { Col: 2, Line: 1, Offset: 1 },
          Roles: ['a', 'c'],
          InternalType: 'bar',
        },
      },
    },
    options: { showLocations: false },
  };

  it('renders correctly', () => {
    const store = mockStore(initialState);
    const node = {
      id: 1,
      InternalType: 'foo',
      Roles: ['e', 'g'],
      StartPosition: { Col: 1, Line: 2, Offset: 3 },
      EndPosition: { Col: 4, Line: 5, Offset: 6 },
      Token: 'bar',
      Properties: {
        a: 'b',
        c: 'd',
        foo: false,
      },
      Children: [2, 3],
    };

    shouldMatchSnapshot(
      <Provider store={store}>
        <Node node={node} showLocations={false} />
      </Provider>
    );
  });

  it('when the node is selected it calls onNodeSelected', () => {
    const spy = jest.fn();
    const node = {
      StartPosition: { Col: 2, Line: 2, Offset: 1 },
      EndPosition: { Col: 6, Line: 2, Offset: 6 },
    };
    const wrapper = mount(<Node node={node} onNodeSelected={spy} />);

    wrapper
      .find(StyledItem)
      .first()
      .simulate('mousemove');
    expect(spy.mock.calls.length).toBe(1);
    const { line, ch } = spy.mock.calls[0][0];
    expect(line).toBe(1);
    expect(ch).toBe(1);
  });

  it('when showLocations is false, it does not display the locations', () => {
    const node = {
      StartPosition: { Col: 2, Line: 2, Offset: 1 },
      EndPosition: { Col: 6, Line: 2, Offset: 6 },
    };
    const wrapper = mount(<Node node={node} showLocations={false} />);

    expect(wrapper.find(Position).length).toBe(0);
  });

  it('when showLocations is true, it does display them', () => {
    const node = {
      StartPosition: { Col: 2, Line: 2, Offset: 1 },
      EndPosition: { Col: 6, Line: 2, Offset: 6 },
    };
    const wrapper = mount(<Node node={node} showLocations={true} />);

    expect(wrapper.find(Position).length).toBe(2);
  });
});
