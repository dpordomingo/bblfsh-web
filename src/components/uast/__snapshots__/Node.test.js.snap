// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Children renders correctly 1`] = `
.jQbgOF {
  margin-left: 20px;
  min-width: 400px;
  background: none;
}

.hRiJmB {
  min-height: 29px;
}

.hRiJmB::before {
  content: '';
  width: 15px;
  display: inline-block;
}

.hbfmnM {
  min-height: 29px;
  cursor: pointer;
}

.hbfmnM::before {
  content: '';
  width: 15px;
  display: inline-block;
}

.hbfmnM::before {
  content: '+';
  color: #EB7422;
}

.hbfmnM:hover > span {
  border-bottom: 1px dashed #B1B1B1;
}

.ghHrsA {
  min-height: 29px;
  cursor: pointer;
}

.ghHrsA::before {
  content: '';
  width: 15px;
  display: inline-block;
}

.ghHrsA::before {
  content: '-';
  color: #F2A83F;
}

.ghHrsA:hover > span {
  border-bottom: 1px dashed #B1B1B1;
}

.bUwxHt {
  display: none;
  border-left: 1px solid #dad7d0;
  margin-left: 4px;
}

.dJnrFt {
  display: block;
  border-left: 1px solid #dad7d0;
  margin-left: 4px;
}

.gaIEnP {
  display: inline;
  font-family: Hack, monospace;
  color: #B1B1B1;
}

.HnGWi {
  color: #F2A83F;
}

.HnGWi::before {
  content: '\\27';
}

.HnGWi::after {
  content: '\\27';
}

.iOQEqP {
  margin-right: 5px;
}

.iOQEqP::after {
  content: ':';
  color: #B1B1B1;
}

<div
  className="sc-bdVaJa jQbgOF"
  onMouseMove={undefined}
>
  <div
    className="sc-htpNat ghHrsA"
    onClick={[Function]}
  >
    <span
      className="sc-bZQynM iOQEqP"
    >
      children
    </span>
    <summary
      className="sc-ifAKCX gaIEnP"
    >
      []Node
    </summary>
  </div>
  <div
    className="sc-bxivhb dJnrFt"
  >
    <div
      className="sc-bdVaJa jQbgOF"
      onMouseMove={[Function]}
    >
      <div
        className="sc-htpNat hbfmnM"
        onClick={[Function]}
      >
        <summary
          className="sc-ifAKCX gaIEnP"
        >
          Node
        </summary>
      </div>
      <div
        className="sc-bxivhb bUwxHt"
      >
        <div
          className="sc-bdVaJa jQbgOF"
        >
          <div
            className="sc-bwzfXH hRiJmB"
          >
            <span
              className="sc-bZQynM iOQEqP"
            >
              internal_type
            </span>
            <span
              className="sc-EHOje HnGWi"
              type="string"
            >
              foo
            </span>
          </div>
        </div>
        <div
          className="sc-bdVaJa jQbgOF"
          onMouseMove={undefined}
        >
          <div
            className="sc-htpNat ghHrsA"
            onClick={[Function]}
          >
            <span
              className="sc-bZQynM iOQEqP"
            >
              roles
            </span>
            <summary
              className="sc-ifAKCX gaIEnP"
            >
              []Role
            </summary>
          </div>
          <div
            className="sc-bxivhb dJnrFt"
          >
            <div
              className="sc-bdVaJa jQbgOF"
            >
              <div
                className="sc-bwzfXH hRiJmB"
              >
                <span
                  className="sc-EHOje HnGWi"
                  type="string"
                >
                  a
                </span>
              </div>
            </div>
            <div
              className="sc-bdVaJa jQbgOF"
            >
              <div
                className="sc-bwzfXH hRiJmB"
              >
                <span
                  className="sc-EHOje HnGWi"
                  type="string"
                >
                  b
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-bdVaJa jQbgOF"
      onMouseMove={[Function]}
    >
      <div
        className="sc-htpNat hbfmnM"
        onClick={[Function]}
      >
        <summary
          className="sc-ifAKCX gaIEnP"
        >
          Node
        </summary>
      </div>
      <div
        className="sc-bxivhb bUwxHt"
      >
        <div
          className="sc-bdVaJa jQbgOF"
        >
          <div
            className="sc-bwzfXH hRiJmB"
          >
            <span
              className="sc-bZQynM iOQEqP"
            >
              internal_type
            </span>
            <span
              className="sc-EHOje HnGWi"
              type="string"
            >
              bar
            </span>
          </div>
        </div>
        <div
          className="sc-bdVaJa jQbgOF"
          onMouseMove={undefined}
        >
          <div
            className="sc-htpNat ghHrsA"
            onClick={[Function]}
          >
            <span
              className="sc-bZQynM iOQEqP"
            >
              roles
            </span>
            <summary
              className="sc-ifAKCX gaIEnP"
            >
              []Role
            </summary>
          </div>
          <div
            className="sc-bxivhb dJnrFt"
          >
            <div
              className="sc-bdVaJa jQbgOF"
            >
              <div
                className="sc-bwzfXH hRiJmB"
              >
                <span
                  className="sc-EHOje HnGWi"
                  type="string"
                >
                  a
                </span>
              </div>
            </div>
            <div
              className="sc-bdVaJa jQbgOF"
            >
              <div
                className="sc-bwzfXH hRiJmB"
              >
                <span
                  className="sc-EHOje HnGWi"
                  type="string"
                >
                  c
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CollapsibleItem renders correctly 1`] = `
.jQbgOF {
  margin-left: 20px;
  min-width: 400px;
  background: none;
}

.ghHrsA {
  min-height: 29px;
  cursor: pointer;
}

.ghHrsA::before {
  content: '';
  width: 15px;
  display: inline-block;
}

.ghHrsA::before {
  content: '-';
  color: #F2A83F;
}

.ghHrsA:hover > span {
  border-bottom: 1px dashed #B1B1B1;
}

.dJnrFt {
  display: block;
  border-left: 1px solid #dad7d0;
  margin-left: 4px;
}

.gaIEnP {
  display: inline;
  font-family: Hack, monospace;
  color: #B1B1B1;
}

.iOQEqP {
  margin-right: 5px;
}

.iOQEqP::after {
  content: ':';
  color: #B1B1B1;
}

<div
  className="sc-bdVaJa jQbgOF"
  onMouseMove={undefined}
>
  <div
    className="sc-htpNat ghHrsA"
    onClick={[Function]}
  >
    <span
      className="sc-bZQynM iOQEqP"
    >
      foo
    </span>
    <summary
      className="sc-ifAKCX gaIEnP"
    />
  </div>
  <div
    className="sc-bxivhb dJnrFt"
  >
    yada yada
  </div>
</div>
`;

exports[`Node renders correctly 1`] = `
.jQbgOF {
  margin-left: 20px;
  min-width: 400px;
  background: none;
}

.hRiJmB {
  min-height: 29px;
}

.hRiJmB::before {
  content: '';
  width: 15px;
  display: inline-block;
}

.hbfmnM {
  min-height: 29px;
  cursor: pointer;
}

.hbfmnM::before {
  content: '';
  width: 15px;
  display: inline-block;
}

.hbfmnM::before {
  content: '+';
  color: #EB7422;
}

.hbfmnM:hover > span {
  border-bottom: 1px dashed #B1B1B1;
}

.ghHrsA {
  min-height: 29px;
  cursor: pointer;
}

.ghHrsA::before {
  content: '';
  width: 15px;
  display: inline-block;
}

.ghHrsA::before {
  content: '-';
  color: #F2A83F;
}

.ghHrsA:hover > span {
  border-bottom: 1px dashed #B1B1B1;
}

.bUwxHt {
  display: none;
  border-left: 1px solid #dad7d0;
  margin-left: 4px;
}

.dJnrFt {
  display: block;
  border-left: 1px solid #dad7d0;
  margin-left: 4px;
}

.gaIEnP {
  display: inline;
  font-family: Hack, monospace;
  color: #B1B1B1;
}

.HnGWi {
  color: #F2A83F;
}

.HnGWi::before {
  content: '\\27';
}

.HnGWi::after {
  content: '\\27';
}

.cihuB {
  color: #212121;
}

.cihuB::before {
  content: ;
}

.cihuB::after {
  content: ;
}

.iOQEqP {
  margin-right: 5px;
}

.iOQEqP::after {
  content: ':';
  color: #B1B1B1;
}

<div
  className="sc-bdVaJa jQbgOF"
  onMouseMove={[Function]}
>
  <div
    className="sc-htpNat hbfmnM"
    onClick={[Function]}
  >
    <summary
      className="sc-ifAKCX gaIEnP"
    >
      Node
    </summary>
  </div>
  <div
    className="sc-bxivhb bUwxHt"
  >
    <div
      className="sc-bdVaJa jQbgOF"
    >
      <div
        className="sc-bwzfXH hRiJmB"
      >
        <span
          className="sc-bZQynM iOQEqP"
        >
          internal_type
        </span>
        <span
          className="sc-EHOje HnGWi"
          type="string"
        >
          foo
        </span>
      </div>
    </div>
    <div
      className="sc-bdVaJa jQbgOF"
      onMouseMove={undefined}
    >
      <div
        className="sc-htpNat ghHrsA"
        onClick={[Function]}
      >
        <span
          className="sc-bZQynM iOQEqP"
        >
          properties
        </span>
        <summary
          className="sc-ifAKCX gaIEnP"
        >
          map&lt;string, string&gt;
        </summary>
      </div>
      <div
        className="sc-bxivhb dJnrFt"
      >
        <div
          className="sc-bdVaJa jQbgOF"
        >
          <div
            className="sc-bwzfXH hRiJmB"
          >
            <span
              className="sc-bZQynM iOQEqP"
            >
              a
            </span>
            <span
              className="sc-EHOje HnGWi"
              type="string"
            >
              b
            </span>
          </div>
        </div>
        <div
          className="sc-bdVaJa jQbgOF"
        >
          <div
            className="sc-bwzfXH hRiJmB"
          >
            <span
              className="sc-bZQynM iOQEqP"
            >
              c
            </span>
            <span
              className="sc-EHOje HnGWi"
              type="string"
            >
              d
            </span>
          </div>
        </div>
        <div
          className="sc-bdVaJa jQbgOF"
        >
          <div
            className="sc-bwzfXH hRiJmB"
          >
            <span
              className="sc-bZQynM iOQEqP"
            >
              foo
            </span>
            <span
              className="sc-EHOje cihuB"
              type="boolean"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-bdVaJa jQbgOF"
      onMouseMove={undefined}
    >
      <div
        className="sc-htpNat ghHrsA"
        onClick={[Function]}
      >
        <span
          className="sc-bZQynM iOQEqP"
        >
          children
        </span>
        <summary
          className="sc-ifAKCX gaIEnP"
        >
          []Node
        </summary>
      </div>
      <div
        className="sc-bxivhb dJnrFt"
      >
        <div
          className="sc-bdVaJa jQbgOF"
          onMouseMove={[Function]}
        >
          <div
            className="sc-htpNat hbfmnM"
            onClick={[Function]}
          >
            <summary
              className="sc-ifAKCX gaIEnP"
            >
              Node
            </summary>
          </div>
          <div
            className="sc-bxivhb bUwxHt"
          >
            <div
              className="sc-bdVaJa jQbgOF"
            >
              <div
                className="sc-bwzfXH hRiJmB"
              >
                <span
                  className="sc-bZQynM iOQEqP"
                >
                  internal_type
                </span>
                <span
                  className="sc-EHOje HnGWi"
                  type="string"
                >
                  foo
                </span>
              </div>
            </div>
            <div
              className="sc-bdVaJa jQbgOF"
              onMouseMove={undefined}
            >
              <div
                className="sc-htpNat ghHrsA"
                onClick={[Function]}
              >
                <span
                  className="sc-bZQynM iOQEqP"
                >
                  roles
                </span>
                <summary
                  className="sc-ifAKCX gaIEnP"
                >
                  []Role
                </summary>
              </div>
              <div
                className="sc-bxivhb dJnrFt"
              >
                <div
                  className="sc-bdVaJa jQbgOF"
                >
                  <div
                    className="sc-bwzfXH hRiJmB"
                  >
                    <span
                      className="sc-EHOje HnGWi"
                      type="string"
                    >
                      a
                    </span>
                  </div>
                </div>
                <div
                  className="sc-bdVaJa jQbgOF"
                >
                  <div
                    className="sc-bwzfXH hRiJmB"
                  >
                    <span
                      className="sc-EHOje HnGWi"
                      type="string"
                    >
                      b
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-bdVaJa jQbgOF"
          onMouseMove={[Function]}
        >
          <div
            className="sc-htpNat hbfmnM"
            onClick={[Function]}
          >
            <summary
              className="sc-ifAKCX gaIEnP"
            >
              Node
            </summary>
          </div>
          <div
            className="sc-bxivhb bUwxHt"
          >
            <div
              className="sc-bdVaJa jQbgOF"
            >
              <div
                className="sc-bwzfXH hRiJmB"
              >
                <span
                  className="sc-bZQynM iOQEqP"
                >
                  internal_type
                </span>
                <span
                  className="sc-EHOje HnGWi"
                  type="string"
                >
                  bar
                </span>
              </div>
            </div>
            <div
              className="sc-bdVaJa jQbgOF"
              onMouseMove={undefined}
            >
              <div
                className="sc-htpNat ghHrsA"
                onClick={[Function]}
              >
                <span
                  className="sc-bZQynM iOQEqP"
                >
                  roles
                </span>
                <summary
                  className="sc-ifAKCX gaIEnP"
                >
                  []Role
                </summary>
              </div>
              <div
                className="sc-bxivhb dJnrFt"
              >
                <div
                  className="sc-bdVaJa jQbgOF"
                >
                  <div
                    className="sc-bwzfXH hRiJmB"
                  >
                    <span
                      className="sc-EHOje HnGWi"
                      type="string"
                    >
                      a
                    </span>
                  </div>
                </div>
                <div
                  className="sc-bdVaJa jQbgOF"
                >
                  <div
                    className="sc-bwzfXH hRiJmB"
                  >
                    <span
                      className="sc-EHOje HnGWi"
                      type="string"
                    >
                      c
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-bdVaJa jQbgOF"
    >
      <div
        className="sc-bwzfXH hRiJmB"
      >
        <span
          className="sc-bZQynM iOQEqP"
        >
          token
        </span>
        <span
          className="sc-EHOje HnGWi"
          type="string"
        >
          bar
        </span>
      </div>
    </div>
    <div
      className="sc-bdVaJa jQbgOF"
      onMouseMove={undefined}
    >
      <div
        className="sc-htpNat ghHrsA"
        onClick={[Function]}
      >
        <span
          className="sc-bZQynM iOQEqP"
        >
          roles
        </span>
        <summary
          className="sc-ifAKCX gaIEnP"
        >
          []Role
        </summary>
      </div>
      <div
        className="sc-bxivhb dJnrFt"
      >
        <div
          className="sc-bdVaJa jQbgOF"
        >
          <div
            className="sc-bwzfXH hRiJmB"
          >
            <span
              className="sc-EHOje HnGWi"
              type="string"
            >
              e
            </span>
          </div>
        </div>
        <div
          className="sc-bdVaJa jQbgOF"
        >
          <div
            className="sc-bwzfXH hRiJmB"
          >
            <span
              className="sc-EHOje HnGWi"
              type="string"
            >
              g
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Position renders correctly 1`] = `
.jQbgOF {
  margin-left: 20px;
  min-width: 400px;
  background: none;
}

.hRiJmB {
  min-height: 29px;
}

.hRiJmB::before {
  content: '';
  width: 15px;
  display: inline-block;
}

.ghHrsA {
  min-height: 29px;
  cursor: pointer;
}

.ghHrsA::before {
  content: '';
  width: 15px;
  display: inline-block;
}

.ghHrsA::before {
  content: '-';
  color: #F2A83F;
}

.ghHrsA:hover > span {
  border-bottom: 1px dashed #B1B1B1;
}

.dJnrFt {
  display: block;
  border-left: 1px solid #dad7d0;
  margin-left: 4px;
}

.gaIEnP {
  display: inline;
  font-family: Hack, monospace;
  color: #B1B1B1;
}

.eTqdey {
  color: #EB7422;
}

.eTqdey::before {
  content: ;
}

.eTqdey::after {
  content: ;
}

.iOQEqP {
  margin-right: 5px;
}

.iOQEqP::after {
  content: ':';
  color: #B1B1B1;
}

<div
  className="sc-bdVaJa jQbgOF"
  onMouseMove={undefined}
>
  <div
    className="sc-htpNat ghHrsA"
    onClick={[Function]}
  >
    <span
      className="sc-bZQynM iOQEqP"
    >
      foo
    </span>
    <summary
      className="sc-ifAKCX gaIEnP"
    >
      Position
    </summary>
  </div>
  <div
    className="sc-bxivhb dJnrFt"
  >
    <div
      className="sc-bdVaJa jQbgOF"
    >
      <div
        className="sc-bwzfXH hRiJmB"
      >
        <span
          className="sc-bZQynM iOQEqP"
        >
          offset
        </span>
        <span
          className="sc-EHOje eTqdey"
          type="number"
        >
          1
        </span>
      </div>
    </div>
    <div
      className="sc-bdVaJa jQbgOF"
    >
      <div
        className="sc-bwzfXH hRiJmB"
      >
        <span
          className="sc-bZQynM iOQEqP"
        >
          line
        </span>
        <span
          className="sc-EHOje eTqdey"
          type="number"
        >
          1
        </span>
      </div>
    </div>
    <div
      className="sc-bdVaJa jQbgOF"
    >
      <div
        className="sc-bwzfXH hRiJmB"
      >
        <span
          className="sc-bZQynM iOQEqP"
        >
          col
        </span>
        <span
          className="sc-EHOje eTqdey"
          type="number"
        >
          1
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Properties renders correctly 1`] = `
.jQbgOF {
  margin-left: 20px;
  min-width: 400px;
  background: none;
}

.hRiJmB {
  min-height: 29px;
}

.hRiJmB::before {
  content: '';
  width: 15px;
  display: inline-block;
}

.ghHrsA {
  min-height: 29px;
  cursor: pointer;
}

.ghHrsA::before {
  content: '';
  width: 15px;
  display: inline-block;
}

.ghHrsA::before {
  content: '-';
  color: #F2A83F;
}

.ghHrsA:hover > span {
  border-bottom: 1px dashed #B1B1B1;
}

.dJnrFt {
  display: block;
  border-left: 1px solid #dad7d0;
  margin-left: 4px;
}

.gaIEnP {
  display: inline;
  font-family: Hack, monospace;
  color: #B1B1B1;
}

.HnGWi {
  color: #F2A83F;
}

.HnGWi::before {
  content: '\\27';
}

.HnGWi::after {
  content: '\\27';
}

.iOQEqP {
  margin-right: 5px;
}

.iOQEqP::after {
  content: ':';
  color: #B1B1B1;
}

<div
  className="sc-bdVaJa jQbgOF"
  onMouseMove={undefined}
>
  <div
    className="sc-htpNat ghHrsA"
    onClick={[Function]}
  >
    <span
      className="sc-bZQynM iOQEqP"
    >
      properties
    </span>
    <summary
      className="sc-ifAKCX gaIEnP"
    >
      map&lt;string, string&gt;
    </summary>
  </div>
  <div
    className="sc-bxivhb dJnrFt"
  >
    <div
      className="sc-bdVaJa jQbgOF"
    >
      <div
        className="sc-bwzfXH hRiJmB"
      >
        <span
          className="sc-bZQynM iOQEqP"
        >
          foo
        </span>
        <span
          className="sc-EHOje HnGWi"
          type="string"
        >
          bar
        </span>
      </div>
    </div>
    <div
      className="sc-bdVaJa jQbgOF"
    >
      <div
        className="sc-bwzfXH hRiJmB"
      >
        <span
          className="sc-bZQynM iOQEqP"
        >
          baz
        </span>
        <span
          className="sc-EHOje HnGWi"
          type="string"
        >
          qux
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Property renders correctly 1`] = `
.jQbgOF {
  margin-left: 20px;
  min-width: 400px;
  background: none;
}

.hRiJmB {
  min-height: 29px;
}

.hRiJmB::before {
  content: '';
  width: 15px;
  display: inline-block;
}

.HnGWi {
  color: #F2A83F;
}

.HnGWi::before {
  content: '\\27';
}

.HnGWi::after {
  content: '\\27';
}

.iOQEqP {
  margin-right: 5px;
}

.iOQEqP::after {
  content: ':';
  color: #B1B1B1;
}

<div
  className="sc-bdVaJa jQbgOF"
>
  <div
    className="sc-bwzfXH hRiJmB"
  >
    <span
      className="sc-bZQynM iOQEqP"
    >
      foo
    </span>
    <span
      className="sc-EHOje HnGWi"
      type="string"
    >
      bar
    </span>
  </div>
</div>
`;

exports[`PropertyName renders correctly 1`] = `
.iOQEqP {
  margin-right: 5px;
}

.iOQEqP::after {
  content: ':';
  color: #B1B1B1;
}

<span
  className="sc-bZQynM iOQEqP"
>
  foo
</span>
`;

exports[`Roles renders correctly 1`] = `
.jQbgOF {
  margin-left: 20px;
  min-width: 400px;
  background: none;
}

.hRiJmB {
  min-height: 29px;
}

.hRiJmB::before {
  content: '';
  width: 15px;
  display: inline-block;
}

.ghHrsA {
  min-height: 29px;
  cursor: pointer;
}

.ghHrsA::before {
  content: '';
  width: 15px;
  display: inline-block;
}

.ghHrsA::before {
  content: '-';
  color: #F2A83F;
}

.ghHrsA:hover > span {
  border-bottom: 1px dashed #B1B1B1;
}

.dJnrFt {
  display: block;
  border-left: 1px solid #dad7d0;
  margin-left: 4px;
}

.gaIEnP {
  display: inline;
  font-family: Hack, monospace;
  color: #B1B1B1;
}

.HnGWi {
  color: #F2A83F;
}

.HnGWi::before {
  content: '\\27';
}

.HnGWi::after {
  content: '\\27';
}

.iOQEqP {
  margin-right: 5px;
}

.iOQEqP::after {
  content: ':';
  color: #B1B1B1;
}

<div
  className="sc-bdVaJa jQbgOF"
  onMouseMove={undefined}
>
  <div
    className="sc-htpNat ghHrsA"
    onClick={[Function]}
  >
    <span
      className="sc-bZQynM iOQEqP"
    >
      roles
    </span>
    <summary
      className="sc-ifAKCX gaIEnP"
    >
      []Role
    </summary>
  </div>
  <div
    className="sc-bxivhb dJnrFt"
  >
    <div
      className="sc-bdVaJa jQbgOF"
    >
      <div
        className="sc-bwzfXH hRiJmB"
      >
        <span
          className="sc-EHOje HnGWi"
          type="string"
        >
          a
        </span>
      </div>
    </div>
    <div
      className="sc-bdVaJa jQbgOF"
    >
      <div
        className="sc-bwzfXH hRiJmB"
      >
        <span
          className="sc-EHOje HnGWi"
          type="string"
        >
          b
        </span>
      </div>
    </div>
    <div
      className="sc-bdVaJa jQbgOF"
    >
      <div
        className="sc-bwzfXH hRiJmB"
      >
        <span
          className="sc-EHOje HnGWi"
          type="string"
        >
          c
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Value should render correctly with value 1`] = `
.eTqdey {
  color: #EB7422;
}

.eTqdey::before {
  content: ;
}

.eTqdey::after {
  content: ;
}

<span
  className="sc-EHOje eTqdey"
  type="number"
>
  1
</span>
`;
